doc/docs.md

```diff
@@ -3806,9 +3806,11 @@ re-creating the original structure exactly.
 Alternatively, you may [embed](#embedded-structs) the sub-data-structures to maintain
 a parallel code structure.
 
-## Debugging generated C code
+## Debugging
 
-To debug issues in the generated C code, you can pass these flags:
+### C code / Native Backend 
```

Comment:
That is slightly misleading. 

I understand that there is a point of view, that only the generated binary matters, so in that sense the C backend is also native, but in V's context, native backend has a specific meaning - i.e. the backend that generates machine code directly, without going through a C intermediary stage and compiling it with a C compiler. Unfortunately, right now, that backend is still not as advanced as the C backend is - it does not support the full V language, and it does not support many of the command line options too.

For example, the V native backend, currently does nothing for -cg and -g. Only the C backend does:
![image](https://user-images.githubusercontent.com/26967/120555708-23889900-c404-11eb-8575-8133e8568a0c.png)

I think it is better to have accurate documentation, that reflects the current state as much as possible, otherwise people tend to read something, try it, and then when it does not work, become disappointed :-| .

Reply:
oh - I understand - but your comment also shows that the term "backend" and the different capabilities of the backends is currently not part of the documentation at all.

Until now I assumed based on reading all exiting documentation that the default backend creating binaries when not specified is "Native" - but based on you feedback above I assume it is "C"?

So I suggest this Structure:

- C Backend binary (Default)
  existing features
- Native Backend binary
  no debugging 
- JS Backend

I removed WASM:
 - the whole process on how to create WASM should be documented in its own file
 - debugging does not work - at least I was not able to get it working based on my own tests

Reply:
The default backend for now is C.  It will be switched to default to native once native is more advanced.  And there's no reason native couldn't support debugging information, too.